cid/common.py

```diff
@@ -1287,8 +1287,9 @@ def create_or_update_dataset(self, dataset_definition: dict, dataset_id: str=Non
             else:
                 print(f'No update requested for dataset {compiled_dataset.get("DataSetId")} {compiled_dataset.get("Name")}={found_dataset.name} ')
         else:
-            self.qs.create_dataset(compiled_dataset)
-
+            dataset_id = self.qs.create_dataset(compiled_dataset)
+            if dataset_id is not None and len(dataset_id) > 0 and compiled_dataset.get("ImportMode") == "SPICE":
```

Comment:
if dataset_id and compiled_dataset.get("ImportMode") == "SPICE":

Reply:
fixed